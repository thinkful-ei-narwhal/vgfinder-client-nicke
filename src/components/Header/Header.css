/* Top-Nav CSS */
/* Add a black background color to the top navigation */
.topnav {
  background-color: #333;
  overflow: hidden;
}

/* Style the links inside the navigation bar */
.topnav a {
  float: left;
  color: #f2f2f2;
  width: 100%;
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
  font-size: 17px;
}

/* Change the color of links on hover */
.topnav a:hover {
  background-color: #ddd;
  color: black;
}

/* Add a color to the active/current link */
.topnav a.active {
  background-color: #5dc21e;
  color: white;
}

.top {
  padding-top: 100px;
  padding-bottom: 100px;
}

/* Hamburger */
.Navbar {
  background-color: #333;
  display: flex;
  padding: 16px;
  font-family: sans-serif;
  color: white;
}

.Navbar__Link {
  padding-right: 8px;
}

.Navbar__Items {
  display: flex;
}

.Navbar__Items--right {
  margin-left: auto;
}

.Navbar__Link-toggle {
  display: none;
}

.Navbar__Link-brand {
  color: lightgray;
  font-weight: bolder;
  height: 30px;
}

.logo {
  display: block;
  width: 960px;
  height: 125px;
  background-color: black;
  content: url("./../../images/VGFinderLogo.png");
}

@media only screen and (max-width: 520px) {
  .Navbar__Items,
  .Navbar {
    flex-direction: column;
  }

  .Navbar__Items {
    display: none;
  }

  .Navbar__Items--right {
    margin-left: 0;
  }

  .Navbar__ToggleShow {
    display: flex;
  }

  .Navbar__Link-toggle {
    align-self: flex-end;
    display: initial;
    position: absolute;
    cursor: pointer;
  }
}

@media only screen and (min-width: 520px) {
  .Navbar__Items,
  .Navbar {
    padding: 0px;
  }

  .Navbar__Link-brand {
    display: none;
  }
}
