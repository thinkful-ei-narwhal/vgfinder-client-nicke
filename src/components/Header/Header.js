import React, { Component } from "react";
import { Link } from "react-router-dom";
import TokenService from "../../services/token-service";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faBars } from "@fortawesome/free-solid-svg-icons";
import "./Header.css";

export default class Header extends Component {
  handleLogoutClick = () => {
    TokenService.clearAuthToken();
  };

  renderLogoutLink() {
    return (
      <div>
        <Link onClick={this.handleLogoutClick} to="/">
          Logout
        </Link>
      </div>
    );
  }

  renderLoginLink() {
    return (
      <div>
        <Link className={this.renderActivePage("/login")} to="/login">
          Log in
        </Link>
      </div>
    );
  }

  renderActivePage(path) {
    if (path === this.props.pathName) {
      this.activeLink = path;
      return "active";
    } else {
      return "";
    }
  }

  renderHamburgerPath() {
    let path = this.props.pathName;
    if (!path) {
      return "Game";
    }
    let pathTrimmed = path.split("/")[1];
    pathTrimmed = pathTrimmed.charAt(0).toUpperCase() + pathTrimmed.slice(1);
    if (pathTrimmed === "Homepage") {
      pathTrimmed = "Home";
    }
    return pathTrimmed;
  }

  classToggle() {
    const navs = document.querySelectorAll(".Navbar__Items");
    navs.forEach((nav) => nav.classList.toggle("Navbar__ToggleShow"));
  }

  render() {
    return (
      <>
        <div className="logo"></div>
        <header className="App__header Navbar">
          <div className="Navbar__Link Navbar__Link-brand">
            {this.renderHamburgerPath()}
          </div>
          <div
            className="Navbar__Link Navbar__Link-toggle"
            onClick={this.classToggle}
          >
            <FontAwesomeIcon icon={faBars} />
          </div>
          <nav className="topnav Navbar__Items">
            <div className="Navbar__Link">
              <Link
                className={this.renderActivePage("/homepage")}
                to="/homepage"
              >
                Home
              </Link>
            </div>
            <div className="Navbar__Link">
              <Link className={this.renderActivePage("/search")} to="/search">
                Search
              </Link>
            </div>
            <div className="Navbar__Link">
              <Link
                className={this.renderActivePage(`/wishlist`)}
                to={`/wishlist`}
              >
                Wishlist
              </Link>
            </div>
            <div className="Navbar__Link">
              <Link
                className={this.renderActivePage("/contribute")}
                to="/contribute"
              >
                Contribute
              </Link>
            </div>

            <nav className="Navbar__Items Navbar__Items--right">
              <div className="Navbar__Link">
                {TokenService.hasAuthToken()
                  ? this.renderLogoutLink()
                  : this.renderLoginLink()}
              </div>
            </nav>
          </nav>
        </header>
      </>
    );
  }
}
