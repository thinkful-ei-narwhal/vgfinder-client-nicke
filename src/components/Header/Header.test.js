import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Header from "./Header";

describe(`Header component`, () => {
  it("renders a Header component without supplied props", () => {
    const wrapper = shallow(<Header />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders a Header component with supplied props", () => {
    const testPath = "/homepage";
    const wrapper = shallow(<Header pathName={testPath} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
