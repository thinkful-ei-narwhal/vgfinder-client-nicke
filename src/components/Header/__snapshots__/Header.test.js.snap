// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders a Header component with supplied props 1`] = `
<Fragment>
  <div
    className="logo"
  />
  <header
    className="App__header Navbar"
  >
    <div
      className="Navbar__Link Navbar__Link-brand"
    >
      Home
    </div>
    <div
      className="Navbar__Link Navbar__Link-toggle"
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f0c9",
              "M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z",
            ],
            "iconName": "bars",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </div>
    <nav
      className="topnav Navbar__Items"
    >
      <div
        className="Navbar__Link"
      >
        <Link
          className="active"
          to="/homepage"
        >
          Home
        </Link>
      </div>
      <div
        className="Navbar__Link"
      >
        <Link
          className=""
          to="/search"
        >
          Search
        </Link>
      </div>
      <div
        className="Navbar__Link"
      >
        <Link
          className=""
          to="/wishlist"
        >
          Wishlist
        </Link>
      </div>
      <div
        className="Navbar__Link"
      >
        <Link
          className=""
          to="/contribute"
        >
          Contribute
        </Link>
      </div>
      <nav
        className="Navbar__Items Navbar__Items--right"
      >
        <div
          className="Navbar__Link"
        >
          <div>
            <Link
              className=""
              to="/login"
            >
              Log in
            </Link>
          </div>
        </div>
      </nav>
    </nav>
  </header>
</Fragment>
`;

exports[`Header component renders a Header component without supplied props 1`] = `
<Fragment>
  <div
    className="logo"
  />
  <header
    className="App__header Navbar"
  >
    <div
      className="Navbar__Link Navbar__Link-brand"
    >
      Game
    </div>
    <div
      className="Navbar__Link Navbar__Link-toggle"
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f0c9",
              "M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z",
            ],
            "iconName": "bars",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </div>
    <nav
      className="topnav Navbar__Items"
    >
      <div
        className="Navbar__Link"
      >
        <Link
          className=""
          to="/homepage"
        >
          Home
        </Link>
      </div>
      <div
        className="Navbar__Link"
      >
        <Link
          className=""
          to="/search"
        >
          Search
        </Link>
      </div>
      <div
        className="Navbar__Link"
      >
        <Link
          className=""
          to="/wishlist"
        >
          Wishlist
        </Link>
      </div>
      <div
        className="Navbar__Link"
      >
        <Link
          className=""
          to="/contribute"
        >
          Contribute
        </Link>
      </div>
      <nav
        className="Navbar__Items Navbar__Items--right"
      >
        <div
          className="Navbar__Link"
        >
          <div>
            <Link
              className=""
              to="/login"
            >
              Log in
            </Link>
          </div>
        </div>
      </nav>
    </nav>
  </header>
</Fragment>
`;
